db/database.go

```diff
@@ -809,58 +817,59 @@ func (db *DatabaseContext) DeleteUserSessions(userName string) error {
 // When compact is run, Sync Gateway initiates a normal delete operation for the document and xattr (a Sync Gateway purge).  This triggers
 // removal of the document from the index.  In the event that the document has already been purged by server, we need to recreate and delete
 // the document to accomplish the same result.
-func (db *Database) Compact() (int, error) {
+func (context *DatabaseContext) Compact() (int, error) {
 
 	// Compact should be a no-op if not running w/ xattrs
-	if !db.UseXattrs() {
+	if !context.UseXattrs() {
 		return 0, nil
 	}
 
 	// Trigger view compaction for all tombstoned documents older than the purge interval
-	purgeIntervalDuration := time.Duration(-db.PurgeInterval) * time.Hour
+	purgeIntervalDuration := time.Duration(-context.PurgeInterval) * time.Hour
 	startTime := time.Now()
 	purgeOlderThan := startTime.Add(purgeIntervalDuration)
-	results, err := db.QueryTombstones(purgeOlderThan)
+	results, err := context.QueryTombstones(purgeOlderThan)
 	if err != nil {
 		return 0, err
 	}
 
-	base.Infof(base.KeyAll, "Compacting purged tombstones for %s ...", base.UD(db.Name))
```

Comment:
Dropped `UD` for Database name here - long term, this should be `MD`. Shamelessly kicking the can a bit.